libraries/AP_RangeFinder/AP_RangeFinder_VL53L0X.cpp

```diff
@@ -245,11 +245,16 @@ AP_RangeFinder_Backend *AP_RangeFinder_VL53L0X::detect(RangeFinder::RangeFinder_
             delete sensor;
             return nullptr;
         }
-        sensor->dev->get_semaphore()->give();
     }
 
-    sensor->init();
+    if (!sensor->init()) {
```

Comment:
This could have been put right after check_id: `if (!sensor->check_id() || !sensor->init())`, but what I don't like is the assumption that semaphore is taken - I know with current code it does because of blocking take, but a small change from 0 to 1 and this code might have the same issue as before.